nifi-frontend/src/main/frontend/apps/nifi-registry/src/app/state/buckets/buckets.effects.ts

```diff
@@ -16,44 +16,191 @@
+
+    createBucket$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(BucketsActions.createBucket),
+            switchMap(({ request, keepDialogOpen }) =>
+                from(this.bucketsService.createBucket(request)).pipe(
+                    map((bucket) => BucketsActions.createBucketSuccess({ response: bucket, keepDialogOpen })),
+                    catchError((errorResponse: HttpErrorResponse) =>
+                        of(
+                            BucketsActions.createBucketFailure(),
+                            this.bannerError(errorResponse, ErrorContextKey.CREATE_BUCKET)
+                        )
+                    )
+                )
+            )
+        )
+    );
+
+    createBucketSuccess$ = createEffect(
+        () =>
+            this.actions$.pipe(
+                ofType(BucketsActions.createBucketSuccess),
+                tap(({ keepDialogOpen }) => {
+                    if (!keepDialogOpen) {
+                        this.dialog.closeAll();
+                    }
+                })
+            ),
+        { dispatch: false }
+    );
+
+    openEditBucketDialog$ = createEffect(
+        () =>
+            this.actions$.pipe(
+                ofType(BucketsActions.openEditBucketDialog),
+                tap(({ request }) => {
+                    this.dialog.open(EditBucketDialogComponent, {
+                        ...MEDIUM_DIALOG,
+                        autoFocus: false,
+                        data: { bucket: request.bucket }
+                    });
+                })
+            ),
+        { dispatch: false }
+    );
+
+    updateBucket$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(BucketsActions.updateBucket),
+            switchMap(({ request }) =>
+                from(this.bucketsService.updateBucket(request)).pipe(
+                    map((bucket) => BucketsActions.updateBucketSuccess({ response: bucket })),
+                    catchError((errorResponse: HttpErrorResponse) =>
+                        of(
+                            BucketsActions.updateBucketFailure(),
+                            this.bannerError(errorResponse, ErrorContextKey.UPDATE_BUCKET)
+                        )
+                    )
+                )
+            )
+        )
+    );
+
+    updateBucketSuccess$ = createEffect(
+        () =>
+            this.actions$.pipe(
+                ofType(BucketsActions.updateBucketSuccess),
+                tap(() => this.dialog.closeAll())
+            ),
+        { dispatch: false }
+    );
+
+    openDeleteBucketDialog$ = createEffect(
+        () =>
+            this.actions$.pipe(
+                ofType(BucketsActions.openDeleteBucketDialog),
+                tap(({ request }) => {
+                    this.dialog.open(DeleteBucketDialogComponent, {
+                        ...MEDIUM_DIALOG,
+                        autoFocus: false,
+                        data: { bucket: request.bucket }
+                    });
```

Comment:
I'm curious why you implemented a new dialog for this delete confirmation rather than using the YesNoDialog from shared? Other places in NiFi use that when confirming a destructive action.

I also question the use of a Banner in this dialog to show an error when deleting. In NiFi, these are typically bubbled up after the confirmation dialog closes in a snackbar or other mechanism. The pattern has typically need to keep a dialog open after a failure if there is some user input that might change the outcome if re-submitted (like create/edit if there is an invalid value or something)

Reply:
I also updated the delete resource UX to use the Yes/No dialog and snackbar error.